sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -62,6 +62,32 @@ class DataFrameStatSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("randomSplit on reordered partitions") {
+    val n = 600
```

Comment:
sure, that sounds reasonable. The only thing that the old test does that this doesn't is that it tests for multiple sampling seeds. I trust that logic might've been also tested somewhere else in the physical planner as well?